### PR TITLE
Sdks 617 fr proximity location

### PR DESCRIPTION
**Pull request by jhoongo**
Following items were changed to align location collection behavior of iOS SDK to Android and JavaScript SDK:
- Change LocationCollector class to request for authorization when the authorization is not asked yet
- Create FRLocationManager class to manage and collect location and/or authorization 
- FRLocationManager checks Privacy Consent in .plist file of the application to determine if SDK can request for authorization
- If the location permission is denied, or restricted, still proceed with LocationCollector with null
- Exposing some of DeviceCollector classes that were private
- Update all FRProximity test cases to reflect changes
- Allow developers to change location accuracy configuration used by CLLocationManager within FRProximity SDK

### CODE REVIEW

File: FRAuth/FRAuth/Callbacks/MultipleValuesCallback.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ import Foundation
  MultipleValuesCallback is a base Callback implementation that has one or more user input values. Any Callback that accepts multiple values from user interaction without OpenAM's validation with policies may inherit from this class.
  */
 @objc(FRMultipleValuesCallback)
-public class MultipleValuesCallback: Callback {
+open class MultipleValuesCallback: Callback {
```
Comment (rodrigoareis) [inline FRAuth/FRAuth/Callbacks/MultipleValuesCallback.swift:17]: Update year on the header for the modified classes

File: FRAuth/FRAuth/Callbacks/SingleValueCallback.swift [hunk 2]
```diff
@@ -14,7 +14,7 @@ import Foundation
  SingleValueCallback is a base Callback implementation that has single user input value. Any Callback that accepts single value from user interaction without OpenAM's validation with policies may inherit from this class.
  */
 @objc(FRSingleValueCallback)
-public class SingleValueCallback: Callback {
+open class SingleValueCallback: Callback {
```
Comment (rodrigoareis) [inline FRAuth/FRAuth/Callbacks/SingleValueCallback.swift:17]: Update year on the header for the modified classes

File: FRAuth/FRAuth/Callbacks/MultipleValuesCallback.swift (hunk 1 above)
Comment (jhoongo) [reply FRAuth/FRAuth/Callbacks/MultipleValuesCallback.swift:17]: Updated the header, thanks for pointing that out.

File: FRAuth/FRAuth/Callbacks/SingleValueCallback.swift (hunk 2 above)
Comment (jhoongo) [reply FRAuth/FRAuth/Callbacks/SingleValueCallback.swift:17]: Updated

**Pull request merged**